lib/compat/wordpress-6.7/blocks.php

```diff
@@ -43,3 +43,58 @@ function gutenberg_filter_block_type_metadata_settings_allow_variations_php_file
 	return $settings;
 }
 add_filter( 'block_type_metadata_settings', 'gutenberg_filter_block_type_metadata_settings_allow_variations_php_file', 10, 2 );
+
+/**
+ * Filter the query vars (tax_query) for the Query Loop block to support post formats.
```

Comment:
I see three separate terms in this function and filter:

 - query vars
 - tax query
 - post formats

I would have expected the name and description to match, but I'm left wondering what I miss when I read `tax_query` - why is it there? why does it matter? did I miss something?

Perhaps there's a way to focus on what this method _does_ specifically from the outside, unless it's destined to be a place that many query vars are filtered. This example is probably wrong because I don't understand what the goal is well enough.

For example…
```php
/**
 * Adds appropriate `post_format` filters to the query loop if necessary for
 * a given block type; e.g. if the block only exists in a template.
 *
 * @param …
 * …
 */
function gutenberg_set_query_vars_for_post_format_from_block_context( $query, $block ) {
	…
}
```

from reading this code, I think there could be more explanation of the purpose of this function and the relationship between the query vars and the block and its context.

Reply:
Since it is filtering, I absolutely think that `filter` needs to be part of the function name.

Reply:
Are you asking why `$query['tax_query']` is in the code or why `tax_query` is in the description?

Reply:
```
Filters the taxonomy query vars (`$query['tax_query']`) on the query loop block
to include or exclude posts that has a post format assigned.
```

Reply:
I think I understand this better now, and I think some of the confusion stems from the fact that there's just so much in the name.

> Since it is filtering, I absolutely think that filter needs to be part of the function name.

This method itself isn't filtering, is it? It's adding additional specifiers to the loop query based on attributes in the query block.

> Are you asking why $query['tax_query'] is in the code or why tax_query is in the description?

My question was less about `tax_query` and more about harmony between the function's description, name, and code.

---

The new description is clearer to me, someone coming into this code without the context of having worked with it already.

Still, I wonder if this could be even more guiding to an unfamiliar audience, without complicating the public interface with internal details.

```php
/**
 * Adds post format query vars to query loop block's WP_Query when the block's attributes call for them.
 *
 * @see 'query_loop_block_query_vars'
 *
 * …
 */
function gutenberg_add_appropriate_post_format_specifiers_to_query_loop( $existing_wp_query, $query_loop_block ) {
```

Reply:
How about just
`gutenberg_add_format_query_vars_to_query_loop`

I am not sure about using "post_format' since this PR was created because it needed to be separated from the post_format taxonomy query.

Reply:
that name sounds clear and to the point. 👍